test/functional/test_framework/authproxy.py

```diff
@@ -188,7 +188,12 @@ def _get_response(self):
                 {'code': -342, 'message': 'non-JSON HTTP response with \'%i %s\' from server' % (http_response.status, http_response.reason)},
                 http_response.status)
 
-        responsedata = http_response.read().decode('utf8')
+        data = http_response.read()
+        try:
+            responsedata = data.decode('utf8')
+        except UnicodeDecodeError as e:
```

Comment:
It is okay to catch the exception as `UnicodeDecodeError` here, as the type of `data` is going to be `bytes` always?
If `data` were not `bytes`, an error would also occur because `data` would not have a `.decode` method